src/utils/utils.js

```diff
@@ -686,6 +696,20 @@ export function generateAddressFromPrivateKey(privKey) {
   return toChecksumAddress(privateToAddress(Buffer.from(privKey, 'hex')).toString('hex'))
 }
 
+export function toChecksumAddressByChainId(address, chainId) {
+  if (Number.parseInt(chainId, 10) === RSK_MAINNET_CODE || Number.parseInt(chainId, 10) === RSK_TESTNET_CODE) {
```

Comment:
chainId might be hex or numeric, currently it only handles for numeric chainIds, we should handle both cases.